### PR TITLE
Feature/jc19 477 anime list management  [React]

### PR DESCRIPTION
**Pull request by andrey-gazzaev**
#JC19-477

### CODE REVIEW

File: apps/react/src/features/anime/components/AnimeList/AnimeList.tsx [hunk 1]
```diff
@@ -92,7 +92,15 @@ const AnimeListComponent: FC<Props> = ({ className }) => {
   }, []);
 
   return (
-    <Paper className={`${styles['anime-catalog']} ${className}`}>
+    <Paper className={styles['animeCatalog']}>
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime/components/AnimeList/AnimeList.tsx:95]: I think you can do it more simple `styles.animeCatalog`

**Review by golovachev-yaroslav — APPROVED** [review]

File: apps/react/src/features/anime/components/ToggleInput/ToggleInput.module.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.toggle {
+  display: flex;
+  align-items: center;
+  border-radius: 5px;
+  background: #f5f5f5;
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/anime/components/ToggleInput/ToggleInput.module.css:5]: Please move color values into variables.

**Review by artem-sotnichenko-saritasa — APPROVED** [review]
I think you have problems with UX. The inputs are small, and look too packed. "Search" and "type" buttons are unnecessary tall, because "sort" is tall. To fix it, you could place them on new lines, use tabs, or use popup.
Also, if select only type and sort, block is overflowing.

File: apps/react/src/api/services/animeService.ts [hunk 3]
```diff
@@ -11,9 +13,15 @@ export namespace AnimeService {
...
   let animeListNextUrl: URL | null = null;
 
-  /** Fetches anime list. */
-  export async function fetchAnimeList(): Promise<readonly AnimeBase[]> {
-    const { data } = await http.get<PaginationDto<AnimeBaseDto>>(animeListUrl.toString());
+  /**
+   * Requests to the server to get anime.
+   * @param animeListQueryParams Parameters for generating a request.
+   */
+  export async function fetchAnimeList(animeListQueryParams: AnimeListQueryParams): Promise<readonly AnimeBase[]> {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/api/services/animeService.ts:20]: Do not force to users to work with `readonly` against their will.

File: apps/react/src/features/anime/components/AnimeList/AnimeList.tsx [hunk 4]
```diff
@@ -1,53 +1,138 @@
...
+            <InfiniteScroll
+              dataLength={animeList.length}
+              next={getMoreAnime}
+              hasMore={true}
+              loader={<h4>Loading...</h4>}
+              scrollableTarget="scrollableDiv"
+            >
+
+              {animeList.map(anime => (
+                <React.Fragment key={anime.id} >
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/anime/components/AnimeList/AnimeList.tsx:125]: No need to import `React` you can import `Fragment` from `react`

File: apps/react/src/features/anime/components/QueryBar/QueryBar.tsx [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        types,
+        sort: { field: sortField, direction: sortDirection },
+      },
+    );
+  }, [search, types, sortField, sortDirection]);
+
+  /** Handles tab panel change. */
+  const handleTabChange = useCallback((event: React.SyntheticEvent, newTabNumber: number) => {
+    setTabNumber(newTabNumber);
+
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/anime/components/QueryBar/QueryBar.tsx:125]: Extra empy line

File: apps/react/src/features/anime/components/QueryBar/QueryBar.tsx [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        ...initialQuery,
+        search,
+        types,
+        sort: { field: sortField, direction: sortDirection },
+      },
+    );
+  }, [search, types, sortField, sortDirection]);
+
+  /** Handles tab panel change. */
+  const handleTabChange = useCallback((event: React.SyntheticEvent, newTabNumber: number) => {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/anime/components/QueryBar/QueryBar.tsx:123]:
```suggestion
  const handleTabChange = useCallback((event: SyntheticEvent, newTabNumber: number) => {
```

File: apps/react/src/features/anime/components/QueryBar/QueryBar.tsx [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+  const [types, setTypes] = useState<readonly AnimeType[]>(initialQuery.types);
+  const [search, setSearch] = useState(initialQuery.search);
+  const [sortField, setSortField] = useState<AnimeSortField>(initialQuery.sort.field);
+  const [sortDirection, setSortDirection] = useState<AnimeSortDirection>(initialQuery.sort.direction);
+
+  /**
+   * Handles search change.
+   * @param event React change event.
+   */
+  const handleSearchChange = useCallback((event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/anime/components/QueryBar/QueryBar.tsx:78]:
```suggestion
  const handleSearchChange = useCallback((event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
```

File: apps/react/src/features/anime/components/QueryBar/QueryBar.tsx [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+          >
+            {sortList.map(item =>
+              <MenuItem
+                key={item.field}
+                value={item.field}
+              >
+                {item.title}
+              </MenuItem>)}
+          </Select>
+          <Button onClick={handleDirectionToggle}>
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/anime/components/QueryBar/QueryBar.tsx:191]: Missing `type` attribute

File: apps/react/src/store/anime/dispatchers.ts [hunk 9]
```diff
@@ -17,3 +19,8 @@ export const fetchNextAnimeList = createAsyncThunk(
     return animeList;
   },
 );
+
+export const removeAnimeList = createAsyncThunk(
+  'anime/remove',
+  () => null,
+);
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/store/anime/dispatchers.ts:26]: There is no need to use thunk for this action, there is no point in such action. You can use `reducer` for this purpose.

File: apps/react/src/store/anime/slice.ts [hunk 10]
```diff
@@ -34,12 +33,13 @@ export const animeSlice = createSlice({
...
-      state.isLoading = false;
     })
     .addCase(fetchNextAnimeList.rejected, (state, action) => {
       if (action.error.message) {
         state.error = action.error.message;
       }
-      state.isLoading = false;
+    })
+    .addCase(removeAnimeList.fulfilled, state => {
+      entityAdapter.removeAll(state as State);
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/store/anime/slice.ts:43]: Let's use reducers over extraReducers here

**Review by timofei-maximov-saritasa — CHANGES_REQUESTED** [review]

**Pull request closed without merging**